### PR TITLE
Golang

### PR DESCRIPTION
**Pull request by shuth-modulr**
Adding in a Golang sample for hmac auth

### CODE REVIEW

**Comment by shuth-modulr** [comment]
whomever merges - can we `squash and merge` please? ta

File: samples/go/hmac/signature/signatureGenerator.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	"net/url"
+)
+
+const (
+	Algorithm       = "algorithm=\"hmac-sha1\","
+	DatePrefix      = "date: "
+	Headers         = "headers=\"date x-mod-nonce\","
+	SignaturePrefix = "signature=\""
+	SignatureSuffix = "\""
+	Nonce           = "\nx-mod-nonce: "
```
Comment (mc1arke) [inline samples/go/hmac/signature/signatureGenerator.go:16]: I'd probably prefer to see this split into `Newline` and `Nonce`

Comment (shuth-modulr) [reply samples/go/hmac/signature/signatureGenerator.go:16]: good shout, will amend

File: samples/go/hmac/signature/signatureGenerator.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+)
+
+const (
+	Algorithm       = "algorithm=\"hmac-sha1\","
+	DatePrefix      = "date: "
+	Headers         = "headers=\"date x-mod-nonce\","
+	SignaturePrefix = "signature=\""
+	SignatureSuffix = "\""
+	Newline 		= "\n"
+	Nonce           = "x-mod-nonce: "
```
Comment (mc1arke) [inline samples/go/hmac/signature/signatureGenerator.go:17]: Your formatting is now inconsistent. Is this intentional?

**Comment by mc1arke** [comment]
Merged. Thanks for the contribution!

**Pull request merged**